src/Microsoft.Health.Fhir.Core/Features/Conformance/Providers/WellKnownConfigurationProvider.cs

```diff
@@ -0,0 +1,122 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Features.Conformance.Models;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Core.Messages.Get;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Core.Features.Conformance.Providers
+{
+    public class WellKnownConfigurationProvider : IWellKnownConfigurationProvider
```

Comment:
If the FHIR server is deployed with a SMART on FHIR compliant identity provider, the SmartOnFhirProxy becomes obsolete because the well-known/smart-configuration endpoint will return the correct configuration.

Reply:
We must support the old SMART on FHIR proxy until 2026.

Reply:
Sure, but should we be adding new functionality to it?

Reply:
Nope - just a note that we need to keep that in mind. 😄